homeassistant/components/motioneye/media_source.py

```diff
@@ -0,0 +1,352 @@
+"""motionEye Media Source Implementation."""
+from __future__ import annotations
+
+import logging
+from pathlib import PurePath
+from typing import Optional, Tuple, cast
+
+from motioneye_client.const import KEY_MEDIA_LIST, KEY_MIME_TYPE, KEY_PATH
+
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_IMAGE,
+    MEDIA_CLASS_MOVIE,
+    MEDIA_CLASS_VIDEO,
+)
+from homeassistant.components.media_source.const import MEDIA_MIME_TYPES
+from homeassistant.components.media_source.error import MediaSourceError, Unresolvable
+from homeassistant.components.media_source.models import (
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import get_media_url, split_motioneye_device_identifier
+from .const import CONF_CLIENT, DOMAIN
+
+MIME_TYPE_MAP = {
+    "movies": "video/mp4",
+    "images": "image/jpeg",
+}
+
+MEDIA_CLASS_MAP = {
+    "movies": MEDIA_CLASS_VIDEO,
+    "images": MEDIA_CLASS_IMAGE,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# Hierarchy:
+#
+# url (e.g. http://my-motioneye-1, http://my-motioneye-2)
+# -> Camera (e.g. "Office", "Kitchen")
+#   -> kind (e.g. Images, Movies)
+#     -> path hierarchy as configured on motionEye
+
+
+async def async_get_media_source(hass: HomeAssistantType) -> MotionEyeMediaSource:
+    """Set up motionEye media source."""
+    return MotionEyeMediaSource(hass)
+
+
+class MotionEyeMediaSource(MediaSource):
+    """Provide motionEye stills and videos as media sources."""
+
+    name: str = "motionEye Media"
+
+    def __init__(self, hass: HomeAssistantType) -> None:
+        """Initialize Xbox source."""
+        super().__init__(DOMAIN)
+        self.hass: HomeAssistantType = hass
```

Comment:
We don't use `HomeAssistantType` anymore.  Just `HomeAssistant` from `homeassistant.core`

Also it shouldn't be needed here since mypy should be able to figure it out since its on the constructor itself.